app/models/miq_alert_status_action.rb

```diff
@@ -0,0 +1,30 @@
+class MiqAlertStatusAction < ApplicationRecord
+  ACTION_TYPES = %w(assign acknowledge comment unassign unacknowledge).freeze
+
+  belongs_to :miq_alert_status
+  belongs_to :assignee, :class_name => 'User'
+  belongs_to :user
+  validates :action_type, :acceptance => { :accept => ACTION_TYPES }, :presence => true
+  validates :user, :presence => true
+  validates :miq_alert_status, :presence => true
+  validates :comment, :presence => true, :if => "action_type == 'comment'"
+  validates :assignee, :presence => true, :if => "action_type == 'assign'"
+  validates :assignee, :absence => true, :unless => "action_type == 'assign'"
+  validate :only_assignee_can_acknowledge
+
+  after_save :update_status_acknowledgement
+
+  def only_assignee_can_acknowledge
+    if action_type == 'acknowledge' && (miq_alert_status.nil? || miq_alert_status.assignee.id != user.id)
+      errors.add(:user, "that is not assigned cannot acknowledge")
+    end
+  end
+
+  def update_status_acknowledgement
+    if %w(assign unassign unacknowledge).include?(action_type)
+      miq_alert_status.update(:acknowledged => false)
+    elsif "acknowledge" == action_type
+      miq_alert_status.update(:acknowledged => true)
+    end
+  end
```

Comment:
If a non-assignee tries "acknowledge", is `only_assignee_can_acknowledge` validated before or after this function?  Could `.update(:acknowledged => true)` write to DB before validation failed?